CRM/Dedupe/MergeHandler.php

```diff
@@ -225,4 +226,35 @@ public function copyDataToNewBlockDAO(string $daoName, $otherBlockId, $name, $bl
     return $otherBlockDAO;
   }
 
+  /**
+   * Get the DAO object appropriate to the location entity.
+   *
+   * @param string $entity
+   *
+   * @return CRM_Core_DAO_Address|CRM_Core_DAO_Email|CRM_Core_DAO_IM|CRM_Core_DAO_Phone|CRM_Core_DAO_Website
+   * @throws \CRM_Core_Exception
+   */
+  public function getDAOForLocationEntity($entity) {
+    switch ($entity) {
+      case 'email':
+        return new CRM_Core_DAO_Email();
+
+      case 'address':
+        return new CRM_Core_DAO_Address();
+
+      case 'phone':
+        return new CRM_Core_DAO_Phone();
+
+      case 'website':
+        return new CRM_Core_DAO_Website();
+
+      case 'im':
+        return new CRM_Core_DAO_IM();
+
+      default:
+        // Mostly here, along with the switch over a more concise format, to help IDEs understand the possibilities.
+        throw new CRM_Core_Exception('Unsupported entity');
+    }
```

Comment:
Hmm, I just removed a long `switch` statement like this in #18546. Aside from the `$entity` being the nonstandard lowercase, is there any reason not to do it this way?

```suggestion
    return CRM_Core_DAO_AllCoreTables::getFullName(CRM_Core_DAO_AllCoreTables::convertEntityNameToCamel($entity, TRUE));
```

Reply:
@colemanw yeah - the follow on patches do more by adding extra variables in there 

As an aside, I think a switch in it's own function is not bad for readability - it's the long switch in the midst of a function that is